purchases/src/main/kotlin/com/revenuecat/purchases/APIKeyValidator.kt

```diff
@@ -0,0 +1,65 @@
+package com.revenuecat.purchases
+
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.strings.ConfigureStrings
+
+private const val GOOGLE_API_KEY_PREFIX = "goog_"
+private const val AMAZON_API_KEY_PREFIX = "amzn_"
+
+internal class APIKeyValidator {
+
+    private enum class APIKeyPlatform {
+        GOOGLE,
+        AMAZON,
+        LEGACY,
+        OTHER_PLATFORM
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```

Comment:
Whoa 🤯

Reply:
This annotation is really useful since it gives you warnings if you try to go against the marked visibility. Though I think we should avoid using it as much as possible, since I believe we should test classes only with their public API. In this case, since the side effects of the public API are only logging, this was much easier though.